### PR TITLE
Fix NIFD to BIDS converter

### LINKED ISSUE #134: A query of the name of the columns in NIFD's  idaSearch_all.csv file
Hello,

I got an Error message when I was trying to convert NIFD dataset to BIDS format (see Pic 1.).
I checked the scripts bylines and I found out that the format of the value (i.e. line[5] in Pic 2. ) seemed not matched to the scripts.
In my idaSearch_all.csv the column in position 5 is "weight", while the value is expected to be related to the date.
May I ask what's the information of the position 5?  Is it the acquired date of the scan? 
(Pic 3. shows the column and the value in the idaSearch I downloaded from the NIFD dataset following the instruction on clinica.run)  

Pic 1: 
image
Pic 2: 
image
Pic 3: 
image

Thank you and hope you have a nice day!
Lily

### LINKED ISSUE #277: Fix nifd-to-bids converter
Some issues have been raised by @14thibea regarding the NIFD-to-Bids converter. An assessment of the validity of the converter is needed. To that end, the following steps are required:

1) Evaluate with bids-validator 
2) Verify that ages correspond to dates 
3) check the absence of  negative values in MMSE

### PR DESCRIPTION
**Pull request by ghisvail**
- Requires `openpyxl` to read the clinical data with Pandas (xlsx format)
- Requires `fsspec` for safe write transactions of the BIDS dataset

Closes #134
Closes #277

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -23,3 +23,5 @@ pydicom
 networkx
 click >= 7
 click-option-group >= 0.5, < 0.6
+openpyxl
```
Comment (mdiazmel) [inline requirements.txt:26]: At some point in the past, we used in Clinica a (deprecated) package called `xlrd` fo read excel files with python. This package (`openpyxl`) seems to have a better support.  I did a quick search of `xlrd` in the code base but I didn't find anything. Could you please remove it from `requirements.txt`?

Comment (ghisvail) [reply requirements.txt:26]: Indeed `xlrd` has the following disclaimer " Please use openpyxl where you can..."

File: clinica/iotools/converters/nifd_to_bids/nifd_utils.py [hunk 2]
```diff
@@ -1,309 +1,327 @@
...
+    import re
+    from pathlib import Path
+
+    pattern = re.compile(
+        r"NIFD_(?:\d_S_\d{4})_(?:\w+)_(?:\d{4}_\d{2}_\d{2})_(?:\d+)_(?:S\d+)_(I\d+)$"
+    )
+
+    for path in Path(imaging_data_directory).rglob("NIFD*"):
+        if path.is_dir():
+            # path = path.relative_to(imaging_data_directory)
```
Comment (mdiazmel) [inline clinica/iotools/converters/nifd_to_bids/nifd_utils.py:104]: Can this line be removed? Or if you consider important to keep it, can you add a comment?

**Review by mdiazmel — COMMENTED** [review]
Thx @ghisvail ,

There are nice ideas, we should try to replicate them in another convertors/pipelines as, for example, the use of `pathlib` to manage file locations. 

I'm not familiar with `fsspec` but I suppose that you're preparing the ground to use Clinica in a cloud environment, which seem good!

I got an error when running the convertor with the current CI data. Here, the last lines of the traceback:
```
File "../clinica/clinica/iotools/converters/nifd_to_bids/nifd_utils.py", line 18, in <lambda>
    .rename_axis(index=lambda x: x.lower().replace(" ", "_"))
AttributeError: 'NoneType' object has no attribute 'lower'
```
Do we need to change the CI data? You need to modify also the non-regression tests (`clinica/test/nonregression/iotools/test_run_converters.py`) according to your modifications.

**Comment by ghisvail** [comment]
> There are nice ideas, we should try to replicate them in another convertors/pipelines as, for example, the use of `pathlib` to manage file locations.

I believe the two main areas of improvement are idiomatic (and therefore efficient) use of Pandas and better path management with pathlib, in that order. More opportunities for refactoring should emerge onwards like better abstractions.

> I'm not familiar with `fsspec` but I suppose that you're preparing the ground to use Clinica in a cloud environment, which seem good!

Indeed. `fspec` provides a unified interface for both local and remote filesystems. That's what powers Pandas feature for reading files from S3 so it sounds solid to me.

> I got an error when running the convertor with the current CI data. Here, the last lines of the traceback:
> 
> ```
> File "../clinica/clinica/iotools/converters/nifd_to_bids/nifd_utils.py", line 18, in <lambda>
>     .rename_axis(index=lambda x: x.lower().replace(" ", "_"))
> AttributeError: 'NoneType' object has no attribute 'lower'
> ```
> 
> Do we need to change the CI data? You need to modify also the non-regression tests (`clinica/test/nonregression/iotools/test_run_converters.py`) according to your modifications.

I think the dataset is in a need for an update. I downloaded a minimal collection (less than 1 GB), following the instructions I updated in the docs. I believe this collection is representative enough of the variety of modalities and use cases present in the dataset.

Could you provide assistance with the CI update process please 🙏

**Comment by mdiazmel** [comment]
> Could you provide assistance with the CI update process please 🙏

Yes, there is a folder location in the "Lustre" where this data can be stored (and the distributed to CI machines). I will share you the location and other details in private.

**Comment by mdiazmel** [comment]
Tested with the dataset!

I'll merge this PR!

**Pull request merged**